pulsar/producer_impl.go

```diff
@@ -124,6 +126,25 @@ func newProducer(client *client, options *ProducerOptions) (*producer, error) {
 		}
 	}
 
+	encryption := options.Encryption
+	// add default message crypto if not provided
+	if encryption != nil && len(encryption.Keys) > 0 {
+		if encryption.KeyReader == nil {
```

Comment:
It seems that there is no similar check in `newConsumer` and `newReader`, do they not need it?

Reply:
Similar check is not needed on consumer/reader  side.